community/kernel/src/main/java/org/neo4j/kernel/impl/store/NeoStore.java

```diff
@@ -178,6 +177,14 @@ protected void triggered( Void event )
                         lastCommittedTx, lastClosedTx ) );
             }
         };
+        try
+        {
+            counts.init();
```

Comment:
If the phases of the lifecycle match up we can. I.e. if `start()` happens after recovery it should be possible. At that point we might be able to add a "rebuild callback" to the counts tracker that we call on `start()` if rebuilding is required, and remove the explicit call to rebuild from the `NeoStore`.